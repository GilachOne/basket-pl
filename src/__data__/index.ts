import oopss from './animations/oopss.json';
import error from './animations/error.json';

export const animations = {
    oopss,
    error,
};
