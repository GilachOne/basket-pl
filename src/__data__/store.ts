import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';

import { api } from './api/main';
import { authApi } from './api/auth';
import { reducer as errorTextReducer } from './slices/errorTextSlice';

export const store = configureStore({
    reducer: {
        [api.reducerPath]: api.reducer,
        [authApi.reducerPath]: authApi.reducer,

        errorText: errorTextReducer,
    },

    middleware: getDefaultMiddleware =>
        getDefaultMiddleware()
            .concat(api.middleware)
            .concat(authApi.middleware),
});

setupListeners(store.dispatch);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
