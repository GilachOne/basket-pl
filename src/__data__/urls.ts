import { getNavigationsValue } from '@ijl/cli';

const baseURL = getNavigationsValue('basket.main');

export const URLs = {
    landing: {
        url: `${baseURL}`,
        isEnabled: getNavigationsValue('basket.main'),
    },
    dashboard: {
        url: `${baseURL}${getNavigationsValue('basket.dashboard')}`,
        isEnabled: getNavigationsValue('basket.dashboard'),
    },
    signIn: {
        url: `${baseURL}${getNavigationsValue('basket.sign.in')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.in'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.in')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.in'),
    },
    signUp: {
        url: `${baseURL}${getNavigationsValue('basket.sign.up')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.up'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.up')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.up'),
    },
    list: {
        url: `${baseURL}${getNavigationsValue('basket.list')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.list')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.list'),
    },
    add: {
        url: `${baseURL}${getNavigationsValue('basket.add')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.add')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.add'),
    },
    resetPassword: {
        url: `${baseURL}${getNavigationsValue('basket.reset.password')}`,
        isEnabled: getNavigationsValue('basket.reset.password'),
    },
};
