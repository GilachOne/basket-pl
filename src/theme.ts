export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
