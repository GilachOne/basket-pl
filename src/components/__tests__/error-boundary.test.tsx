import React from 'react';
import { describe, it, expect } from '@jest/globals';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { BrowserRouter } from 'react-router-dom';

import { theme } from '../../theme';
import { store } from '../../__data__/store';
import ErrorBoundary from '../error-boundary/error-boundary';

// eslint-disable-next-line react/display-name
jest.mock('lottie-react', () => () => <h1>Что-то пошло не так :(</h1>);

describe('ErrorBoundary Component', () => {

    it('ErrorBoundary Test', () => {
        const ThrowError = () => {
            throw new Error('Test');
          };
        const { container } = render(
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <ErrorBoundary>
                            <ThrowError />
                        </ErrorBoundary>
                    </BrowserRouter>
                </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});