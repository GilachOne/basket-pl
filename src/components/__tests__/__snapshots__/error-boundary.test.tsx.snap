// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component ErrorBoundary Test 1`] = `
<div>
  <div
    class="css-1d342t8"
  >
    <h1>
      Что-то пошло не так :(
    </h1>
    <div
      class="css-8428uh"
    >
      <h1>
        Что-то пошло не так :(
      </h1>
    </div>
  </div>
</div>
`;
